test/e2e/e2e_suite_test.go

```diff
@@ -0,0 +1,204 @@
+// In this case, the local clusterctl repository & the bootstrap cluster are shared across all the tests.
+var _ = SynchronizedBeforeSuite(func() []byte {
+	// Before all ParallelNodes
+	artifactPath = os.Getenv("ARTIFACTS")
+
+	By("creating the logs directory")
+	logsPath = path.Join(artifactPath, "logs")
+	Expect(os.MkdirAll(filepath.Dir(logsPath), 0755)).To(Succeed())
+
+	By("creating the providers directory")
+	providerLogsPath := path.Join(logsPath, "common")
+	Expect(os.MkdirAll(filepath.Dir(providerLogsPath), 0755)).To(Succeed())
+
+	By("creating the resources directory")
+	resourcesPath = path.Join(artifactPath, "resources")
+	Expect(os.MkdirAll(filepath.Dir(resourcesPath), 0755)).To(Succeed())
+
+	Expect(configPath).To(BeAnExistingFile(), "invalid argument. e2e.config should be an existing file")
+	Expect(artifactPath).To(BeADirectory(), "invalid argument. e2e.artifacts-folder should be an existing folder")
+
+	By("Initializing a runtime.Scheme with all the GVK relevant for this test")
+	scheme = runtime.NewScheme()
+	framework.TryAddDefaultSchemes(scheme)
+	// TODO make adding infra provider schemes generic (if it is possible)
+	Expect(infrav1.AddToScheme(scheme)).To(Succeed())
+
+	By(fmt.Sprintf("Loading the e2e test configuration from %s", configPath))
+	e2eConfig = clusterctl.LoadE2EConfig(context.TODO(), clusterctl.LoadE2EConfigInput{ConfigPath: configPath})
+	Expect(e2eConfig).ToNot(BeNil(), "Failed to load E2E config")
+
+	By(fmt.Sprintf("Creating a clusterctl local repository into %s", artifactPath))
+	// NB. Creating the local clusterctl repository into the the artifact folder so it will be preserved at the end of the test
+	clusterctlConfigPath = clusterctl.CreateRepository(context.TODO(), clusterctl.CreateRepositoryInput{
+		E2EConfig:     e2eConfig,
+		ArtifactsPath: artifactPath,
+	})
+	Expect(clusterctlConfigPath).To(BeAnExistingFile(), "Failed to get a clusterctl config file")
+
+	By("Initializing a management cluster")
+	//TODO: manage "use existing cluster" scenario. Initial assumption: existing cluster should have providers
+	//TODO: manage "use kind without CAPD" scenario.
+	//TODO: stream logs out of controllers
+	//TODO: define log folder organization (logs for the bootstrap cluster, logs for each spec
+	managementCluster, managementClusterKubeConfigPath = clusterctl.InitManagementCluster(context.TODO(), &clusterctl.InitManagementClusterInput{
+		E2EConfig:            e2eConfig,
+		ClusterctlConfigPath: clusterctlConfigPath,
+		Scheme:               scheme,
+		NewManagementClusterFn: func(name string, scheme *runtime.Scheme) (cluster framework.ManagementCluster, kubeConfigPath string, err error) {
+			cluster, err = capde2e.NewClusterForCAPD(context.TODO(), name, scheme)
+			if err != nil {
+				return nil, "", err
+			}
+			//TODO: this is hacky way to get the kubeconfig path. As soon as v1alpha4 opens, we should add GetKubeConfigPath method to the ManagementCluster interface and get rid of this
```

Comment:
is adding a new method to the framework breaking the contract of v1alpha3?

Reply:
I saw other PRs adding methods to the same interface, so I guess this is acceptable

Reply:
Now there is also the api-diff jobs that cover compatibility issues